Lib/subprocess.py

```diff
@@ -978,6 +979,19 @@ def __init__(self, args, bufsize=-1, executable=None,
 
             raise
 
+    def __repr__(self):
+        max_args_length = 30
+        returncode = self.returncode or 'unfinished'
+        args = ' '.join(map(shlex.quote, self.args))
+
+        if len(args) > max_args_length:
+            args = f"{args[:max_args_length]}...".rstrip()
+
+        return (
+            f"<{self.__module__}.{self.__class__.__name__}: "
+            f"pid:{self.pid} args:{args} returncode:{returncode}>"
```

Comment:
IMO the pid should be removed, see [my comment to this effect on the bpo issue](https://bugs.python.org/issue38724#msg356598).

Also I suggest placing the return code before the args, since the args may be rather long.
```suggestion
            f"returncode:{returncode} args:{args}>"
```